lib/plugins/aws/invokeLocal/index.js

```diff
@@ -177,8 +189,145 @@ class AwsInvokeLocal {
         this.options.context);
     }
 
-    throw new this.serverless.classes
-      .Error('You can only invoke Node.js, Python, Java & Ruby functions locally.');
+    return this.invokeLocalDocker();
+  }
+
+
+  checkDockerImage() {
+    const runtime = this.options.functionObj.runtime
+      || this.serverless.service.provider.runtime
+      || 'nodejs4.3';
+
+    return new BbPromise((resolve, reject) => {
+      const docker = spawn('docker', ['images', '-q', `lambci/lambda:${runtime}`]);
+      let stdout = '';
+      docker.stdout.on('data', (buf) => { stdout += buf.toString(); });
+      docker.on('close', error => (error ? reject(error) : resolve(Boolean(stdout.trim()))));
+    });
+  }
+
+  pullDockerImage() {
+    const runtime = this.options.functionObj.runtime
+      || this.serverless.service.provider.runtime
+      || 'nodejs4.3';
+
+    this.serverless.cli.log('Downloading base docker image...');
```

Comment:
Probably personal preference, but I'd vote for `Docker` vs. `docker` since it's a company / product name.